internal/app/machined/pkg/controllers/cri/registry.go

```diff
@@ -0,0 +1,118 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package cri
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/controller/generic/transform"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/siderolabs/gen/optional"
+	"go.uber.org/zap"
+
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+	"github.com/siderolabs/talos/pkg/machinery/resources/config"
+	"github.com/siderolabs/talos/pkg/machinery/resources/cri"
+)
+
+// ConfigController watches v1alpha1.Config, updates registry.RegistriesConfig.
+type ConfigController = transform.Controller[*config.MachineConfig, *cri.RegistriesConfig]
+
+// NewConfigController instanciates the config controller.
+//
+//nolint:gocyclo
+func NewConfigController() *ConfigController {
```

Comment:
should be probably `RegistriesConfigController` (named after the output resource)

Reply:
Done